applications/luci-app-sshtunnel/htdocs/luci-static/resources/view/sshtunnel.js

```diff
@@ -0,0 +1,403 @@
+		o.rmempty = false;
+
+
+		s = m.section(form.GridSection, "tunnelL", _("Local Tunnels"),
+			_("When the connection initiated at LocalAddress:LocalPort and then forwarded to RemoteAddress:RemotePort")
+		);
+		s.anonymous = true;
+		s.addremove = true;
+		s.nodescriptions = true;
+
+		o = s.option(form.Flag, "enabled", _("Enabled"));
+		o.default = "1";
+
+		o = _addServerOption(s);
+
+		o = s.option(form.Value, "localaddress", _("Local address"),
+			_("* means to accept a connection from any interface on the OpenWrt side")
+		);
+		o.datatype = 'or(host, "*")';
+		o.optional = true;
+		o.rmempty = false;
+
+		o = s.option(form.Value, "localport", _("Local port"), "");
+		o.placeholder = "1022";
+		o.datatype = "port";
+		o.rmempty = false;
+
+		o = s.option(form.Value, "remoteaddress", _("Remote address"), "");
+		o.datatype = "host";
+		o.rmempty = false;
+
+		o = s.option(form.Value, "remoteport", _("Remote port"), "");
+		o.default = "22";
+		o.datatype = "port";
+		o.rmempty = false;
+
+
+		s = m.section(form.GridSection, "tunnelD", _("Dynamic Tunnels"),
+			_("When the connection initiated with the SOCKS protocol to the LocalAddress:LocalPort and then forwarded over the remote host")
+		);
+		s.anonymous = true;
+		s.addremove = true;
+		s.nodescriptions = true;
+
+		o = s.option(form.Flag, "enabled", _("Enabled"));
+		o.default = "1";
+
+		o = _addServerOption(s);
+
+		o = s.option(form.Value, "localaddress", _("Local address"),
+			_("* means to accept a connection from any interface on the OpenWrt side")
+		);
+		o.datatype = 'or(host, "*")';
+		o.optional = true;
+		o.rmempty = false;
+
+		o = s.option(form.Value, "localport", _("Local port"), "");
+		o.default = "1080";
+		o.datatype = "port";
+		o.rmempty = false;
+
+
+		s = m.section(form.GridSection, "tunnelW", _("VPN Tunnels"),
+			_("Creates TUN/TAP devices on client and server to establish a VPN tunnel between them")
+		);
+		s.anonymous = true;
+		s.addremove = true;
+		s.nodescriptions = true;
+
+		o = s.option(form.Flag, "enabled", _("Enabled"));
+		o.default = "1";
+
+		o = _addServerOption(s);
+
+		o = s.option(form.ListValue, "vpntype", _("VPN type"), "");
+		o.value("point-to-point", "TUN (point-to-point)");
+		o.value("ethernet", "TAP (ethernet)");
+		o.default = "point-to-point";
+
+		o = s.option(form.Value, "localdev", _("Local dev"), "");
+		o.default = "any";
+		o.datatype = 'or("any", min(0))';
+		o.rmempty = false;
+
+		o = s.option(form.Value, "remotedev", _("Remote dev"), "");
+		o.default = "any";
+		o.datatype = 'or("any", min(0))';
+		o.rmempty = false;
+
+
+		s = m.section(form.GridSection, "_keys");
+		s.render = L.bind(_renderSshKeys, this, sshKeys);
+
+		s = m.section(form.GridSection, "_known_hosts");
+		s.render = L.bind(_renderKnownHosts, this, knownHosts);
+
+		return m.render();
+	},
+});
+
+function _addServerOption(s) {
+	var o = s.option(form.ListValue, "server", _("Server"), _("Type a server name from servers list above"));
```

Comment:
`var o = s.option(form.ListValue, "server", _("Server"), _("Enter a server name from the server list above."));`

Reply:
fixed